bundles/org.dataflowanalysis.analysis.converter/src/org/dataflowanalysis/analysis/converter/Converter.java

```diff
@@ -0,0 +1,71 @@
+package org.dataflowanalysis.analysis.converter;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.log4j.Logger;
+
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.emf.ecore.resource.Resource;
+import org.eclipse.emf.ecore.resource.ResourceSet;
+import org.eclipse.emf.ecore.resource.impl.ResourceImpl;
+import org.eclipse.emf.ecore.resource.impl.ResourceSetImpl;
+import org.eclipse.emf.ecore.xmi.XMLResource;
+import org.eclipse.emf.ecore.xmi.impl.XMLResourceFactoryImpl;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+public abstract class Converter {
+    protected ObjectMapper objectMapper;
+    protected File file;
+
+    protected final Logger logger = Logger.getLogger(Converter.class);
+
+    public Converter() {
+        objectMapper = new ObjectMapper();
+    }
+
+    public void store(DataFlowDiagramAndDictionary complete, String outputFile) {
+        String truncatedOutputFile;
+        if (outputFile.endsWith(".json")) {
+            truncatedOutputFile = outputFile.substring(0, outputFile.length() - 5);
+        } else {
+            truncatedOutputFile = outputFile;
+        }
+        ResourceSet resourceSet = new ResourceSetImpl();
+        Resource dfdResource = createAndAddResource(truncatedOutputFile + ".dataflowdiagram", new String[] {"dataflowdiagram"}, resourceSet);
+        Resource ddResource = createAndAddResource(truncatedOutputFile + ".datadictionary", new String[] {"datadictionary"}, resourceSet);
+
+        dfdResource.getContents().add(complete.dataFlowDiagram());
+        ddResource.getContents().add(complete.dataDictionary());
+
+        saveResource(dfdResource);
+        saveResource(ddResource);
+
+    }
+
+    private Resource createAndAddResource(String outputFile, String[] fileextensions, ResourceSet resourceSet) {
+        for (String fileextension : fileextensions) {
+            resourceSet.getResourceFactoryRegistry().getExtensionToFactoryMap().put(fileextension, new XMLResourceFactoryImpl());
+        }
+        URI uri = URI.createFileURI(outputFile);
+        Resource resource = resourceSet.createResource(uri);
+        ((ResourceImpl) resource).setIntrinsicIDToEObjectMap(new HashMap<>());
+        return resource;
+    }
+
+    private void saveResource(Resource resource) {
+        Map<Object, Object> saveOptions = ((XMLResource) resource).getDefaultSaveOptions();
+        saveOptions.put(XMLResource.OPTION_CONFIGURATION_CACHE, Boolean.TRUE);
```

Comment:
see outdated comment. Why are you doing this here? Performance is not an issue and this should not be needed. If it is necessary, please add a comment why?

Reply:
Same here, this is from the wiki

Reply:
Ok, you can just remove this, I think. Just look how we load and store models in the analysis. It is much simpler now. This is from a time when loading models was not as optimized as it is now (looooooong time ago).
> I see you are trying to improve performance, but did you actually run into performance issues (meaning minute to hour-long load/save times)? If not please remove.
This converter is not some performance-critical piece of software and readability/understandability is way more important than performance.